safe/gui/tools/peta_jakarta_dialog.py

```diff
@@ -537,3 +528,28 @@ def reject(self):
         # add our own logic here...
 
         super(PetaJakartaDialog, self).reject()
+
+    def download(self, url, output_path):
+        """Download file from API url and write to output path.
+
+        :param url: URL of the API.
+        :type url: str
+
+        :param output_path: Path of output file,
+        :type output_path: str
+        """
+        request_failed_message = self.tr(
+            "Can't access PetaBencana API: {source}").format(
+            source=url)
+        downloader = FileDownloader(url, output_path)
+        result = downloader.download()
+        if not result:
+            raise Exception(request_failed_message)
```

Comment:
@myarjunar We should catch and display a nice error message.
Can you use a simple `display_critical_message_box` with the requested_fieled_message and a title ?

BTW, can also edit the OSM Downloader in the same way ? Or do you want to do it after this PR has been merged ?